docs/developer/style_guide.md

```diff
@@ -1,159 +1,251 @@
 
-> **Note** - When using SQLAlchemy to make query data model, it is best practice to name the code executed (the `select` statement or similar) `statement` or `stmt`. The actual `query` is the result of the statement's execution. 
+### Variable Assignment
 
-### Examples 
-#### Simple Select 
-```python 
-statement = select(
-  models.Table.ColumnA, 
-  models.Table.ColumnB, 
+SQL statements should be assigned to a variable, then executed separately.
+The name `stmt` is a common choice.
+
+```python
+# Wrong
+rses = session.execute(select(models.RSE)).scalars().all()
+
+# Right
+stmt = select(
+    models.RSE
 )
-query = session.execute(statement).scalars()
-for column_a, column_b in query: 
-  ...
+rses = session.execute(stmt).scalars().all()
 ```
-#### Select with Join 
-```python 
-statement = select(
-  models.Table1.ColumnA, 
-  models.Table1.ColumnB, 
-  models.Table2.Column2
-).join(
-  models.Table2, 
-  (models.Table1.keyA == models.Table2.keyA) & (models.Table1.keyB == models.Table2.keyB) 
+
+### SQLAlchemy Syntax
+
+All new code should use the more recent SQLAlchemy 2.0 syntax.
+Existing code using the older 1.4 syntax should be migrated to the 2.0 syntax.
+
+```python
+# Wrong
+rses = (session.query(models.RSE)
+               .all())
+
+# Right
+stmt = select(
+    models.RSE
 )
-query = session.execute(statement).scalars()
-for column_a, column_b, column_c in query: 
-  ...
+rses = session.execute(stmt).scalars().all()
 ```
 
-#### Multiple Conditions
-```python 
-statement = select(
-  models.Table1.Column_a, 
-  models.Table1.Column_b, 
-  models.Table1.Column_c,
+### Whitespace
+
+The functions that return basic SQL constructs (e.g. `select()`, `update()`, and `delete()`) should have a newline after the opening parenthesis and before the closing parenthesis.
+Same applies to all methods of those constructs (e.g. `distinct()`, `join()`, and `where()`).
+The latter should be ordered in a way that matches the syntax of SQL, when permittable.
+Inside the parentheses, each argument should be indented and put on a separate line.
+
+```python
+# Wrong
+stmt = (
+    select(models.RSEAttrAssociation.value)
+    .where(models.RSEAttrAssociation.key == 'fts')
+    .distinct()
+)
+
+# Right
+stmt = select(
+    models.RSEAttrAssociation.value
+).distinct(
 ).where(
-  and_(
-    models.Table1.Condition1 == True,  
-    models.Table1.Condition2 != None, 
-    models.Table1.Condition2 < {Different Statement}
-  )
+    models.RSEAttrAssociation.key == 'fts'
 )
-query = session.execute(statement).scalars()
-for column_a, column_b, column_c in query: 
-  ...
 ```
 
-#### Checking existence/Single Value
-When using the query to ensure that the table has been populated, or only a single result is required, use either `session.execute().one()` or `session.execute().scalar_one()`. 
+### Discouraged Logical Operators
+
+The overloaded operators `&` and `|` should not be used due to their precedence.
```

Comment:
```suggestion
The overloaded operators `&` and `|` should not be used due to their precedence.
The [SQLAlchemy Expression constructs](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#column-element-foundational-constructors) are preferred.
```

Reply:
I agree with this suggestion; the example is definitely clear enough,, but it's even clearer if we describe in words what we should use in place of `&` and `|`.

Reply:
I contemplated it over the weekend and I’m not convinced linking to the SQLAlchemy documentation is the best approach; it would have been acceptable if there were a separate table for logical operators. Instead, I’ve opted to copy the wording of the ‘Python Keywords’ section, which did not raise any comments.